src/lib.rs

```diff
@@ -609,6 +627,18 @@ fn with_rng<R>(f: impl FnOnce(&mut Rng) -> R) -> R {
     })
 }
 
+/// Try to run an operation with the current thread-local generator.
+#[inline]
+fn try_with_rng<R>(f: impl FnOnce(&mut Rng) -> R) -> Result<R, std::thread::AccessError> {
+    RNG.try_with(|rng| {
+        let current = rng.replace(Rng(0));
+
+        let mut restore = RestoreOnDrop { rng, current };
```

Comment:
can we be sure that this doesn't corrupt the `Rng` in case of a panic? or doesn't that matter because the `u64` seed doesn't have invalid states and thus we can just do that anyways? why isn't that an exposed functionality/method of `Cell`?

Reply:
I don't think this is an issue, since with `&mut` there shouldn't be any way to corrupt it with uninitialized data.

Reply:
ok, what is the function of the `RestoreOnDrop` here, why is it necessary?

Reply:
In order to use the `Rng` stored in the thread-local variable, the `Rng` needs to be replaced with a zero value and moved out, then moved back in once the operation is done. The `RestoreOnDrop` guard ensures that it's moved back in even in the event of a panic (which can happen if the range bounds are invalid), preventing the thread-local variable from being set to 0 and causing a loss of entropy.

Reply:
I'd appreciate it if we would simply define the `Rng` to be `Clone` + `Copy`, which would allow us to use more convenient methods on `Cell<Rng>`.